src/Web/XHR.purs

```diff
@@ -0,0 +1,157 @@
+module Web.XHR
+  ( xmlHttpRequest
+  , abort
+  , getAllResponseHeaders
+  , getResponseHeader
+  , open
+  , open'
+  , overrideMimeType
+  , send
+  , sendString
+  , sendDocument
+  , sendBlob
+  , sendFormData
+  , setRequestHeader
+  , readyState
+  , response
+  , responseURL
+  , status
+  , statusText
+  , timeout
+  , setTimeout
+  , upload
+  , withCredentials
+  , setWithCredentials
+  ) where
```

Comment:
Perhaps we should re-export the `Types` stuff that is most useful also? Do we even need a `Types` module for that matter?

Or if this is supposed to be following the `purescript-dom` conventions (such as they are) for module naming this module would be `Web.XHR.XMLHttpRequest`. We could then re-export the most useful bits of both from a new `Web.XHR`.